src/tools/line.js

```diff
@@ -0,0 +1,166 @@
+                    handleOne, handleTwo,
+                    textAttributes = {x: -5, y: -5};
+                switch (orient) {
+                    case 'horizontal':
+                        translation = function(a, b) { return 'translate(' + a + ', ' + b + ')'; };
+                        lineProperty = 'x2';
+                        crossScale = xScale;
+                        valueScale = yScale;
+                        handleOne = 'left-handle';
+                        handleTwo = 'right-handle';
+                        break;
+
+                    case 'vertical':
+                        translation = function(a, b) { return 'translate(' + b + ', ' + a + ')'; };
+                        lineProperty = 'y2';
+                        crossScale = yScale;
+                        valueScale = xScale;
+                        textAttributes.transform = 'rotate(-90)';
+                        handleOne = 'bottom-handle';
+                        handleTwo = 'top-handle';
+                        break;
+
+                    default:
+                        throw new Error('Invalid orientation');
+                }
+
+                // ordinal axes have a rangeExtent function, this adds any padding that
+                // was applied to the range. This functions returns the rangeExtent
+                // if present, or range otherwise
+                function range(scale) {
+                    return scale.rangeExtent ? scale.rangeExtent() : scale.range();
+                }
+
+                var scaleRange = range(crossScale),
+                    // the transform that sets the 'origin' of the annotation
+                    containerTransform = function(d) {
+                        var transform = value(d);
+                        if (domainValue) {
+                            transform = valueScale(transform);
+                        }
+                        return translation(scaleRange[0], transform);
+                    },
+                    scaleWidth = scaleRange[1] - scaleRange[0];
+
+                var container = d3.select(this);
+
+                // Create a group for each line
+                data = Array.isArray(data) ? data : [data];
+                var g = dataJoin(container, data);
+
+                // create the outer container and line
+                var enter = g.enter()
+                    .attr('transform', containerTransform);
+                enter.append('line')
+                    .attr(lineProperty, scaleWidth);
+
+                // create containers at each end of the annotation
+                enter.append('g')
+                    .classed(handleOne, true);
+
+                enter.append('g')
+                    .classed(handleTwo, true)
+                    .attr('transform', translation(scaleWidth, 0))
+                    .append('text')
+                    .attr(textAttributes);
+
+                // Update
```

Comment:
By reading the update will not handle crossScale range changes. Is that intentional?